pandas/core/strings.py

```diff
@@ -1000,16 +1009,20 @@ def _wrap_result(self, result):
 
         if not hasattr(result, 'ndim'):
             return result
-        elif result.ndim == 1:
-            name = getattr(result, 'name', None)
+
+        if 'name' in kwargs:
+            name = kwargs['name']
+        else:
+            name = getattr(result, 'name', None) or self.series.name
```

Comment:
you can just do:

``name = kwargs.get('name') or getattr(result,'name',None) or self.series.name``